pkg/build/controller/build/build_controller.go

```diff
@@ -242,24 +264,35 @@ func (bc *BuildController) Run(workers int, stopCh <-chan struct{}) {
 	defer utilruntime.HandleCrash()
 	defer bc.buildQueue.ShutDown()
 	defer bc.buildConfigQueue.ShutDown()
-
-	// Read additionalCA data, if it exists
-	if len(bc.additionalTrustedCAPath) > 0 {
-		caData, err := bc.readBuildCAData()
-		if err != nil {
-			glog.Warningf("Failed to read additional CA bundle %s: %v", bc.additionalTrustedCAPath, err)
-		}
-		bc.additionalTrustedCAData = caData
-	}
+	defer bc.controllerConfigQueue.ShutDown()
 
 	// Wait for the controller stores to sync before starting any work in this controller.
-	if !cache.WaitForCacheSync(stopCh, bc.buildStoreSynced, bc.podStoreSynced, bc.secretStoreSynced, bc.imageStreamStoreSynced) {
+	if !cache.WaitForCacheSync(stopCh,
+		bc.buildStoreSynced,
+		bc.podStoreSynced,
+		bc.secretStoreSynced,
+		bc.imageStreamStoreSynced,
+		bc.configMapStoreSynced) {
 		utilruntime.HandleError(fmt.Errorf("timed out waiting for caches to sync"))
 		return
 	}
 
+	// Separately wait for the build controller config cache to sync since not all tests for this controller run on installed clusters
+	// If this fails the likely reason is that the CRD for build.config.openshift.io was not created.
+	if !waitForCacheSync(10*time.Second, bc.controllerConfigStoreSynced) {
+		utilruntime.HandleError(fmt.Errorf("timed out waiting for controller config cache to sync"))
+	}
```

Comment:
@bparees this is where you were concerned about failure modes.

If we're only syncing the CA's, the worst case is that builds fail because they can't pull/push to an untrusted registry. This is a safe failure imo.

However, @coreydaley will be building off of this to sync the default proxy. Worst case there is that when builds run, the desired proxy is not used. Possible failure modes:

1. Disconnected/air-gap environments - builds fail doing any action that requires the proxy. "Safe" failure.
2. Connected environments that use a proxy anyway - builds succeed, but may download content in violation of security policies. Is this something we need to worry about?

Reply:
i think i'd rather control this w/ an env var.  only wait for this cache to sync if `INTEGRATION_TEST_ENVIRONMENT=true` is not set.  or something along those lines.

then we can update the integration test launch to set the env var.

Reply:
Also upcoming work for the registry config for search, whitelisted, and blacklisted registries. Latter two would be very bad if we allow the default (allow all registries) when a cluster admin wants registries blacklisted or whitelisted.

Reply:
updated to gate cache sync via an environment variable.